controllers/auth.go

```diff
@@ -617,6 +618,13 @@ func (c *ApiController) Login() {
 				c.ResponseError(fmt.Sprintf(c.T("auth:Failed to login in: %s"), err.Error()))
 				return
 			}
+
+			if provider.EmailRegex != "" {
```

Comment:
First, `provider.EmailRegex` should be validated in UpdateProvider() API

Here you should also return error if the regex is invalid instead of panic. We don't use panic any more.

We should already have such code, copy it